sdk/spring/azure-spring-boot-test-aad-b2c/src/test/java/com/azure/test/aad/b2c/selenium/AADB2CSeleniumITHelper.java

```diff
@@ -36,98 +30,47 @@
-        Process process = null;
-        try {
-            File dir = new File(directory);
-            if (Pattern.matches("linux.*", osName)) {
-                process = Runtime.getRuntime().exec("chmod +x " + chromedriverLinux, null, dir);
-                process.waitFor();
-                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverLinux);
-            } else if (Pattern.matches("windows.*", osName)) {
-                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverWin32);
-            } else if (Pattern.matches("mac.*", osName)) {
-                process = Runtime.getRuntime().exec("chmod +x " + chromedriverMac, null, dir);
-                process.waitFor();
-                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverMac);
-            } else {
-                throw new IllegalStateException("Unrecognized osName. osName = " + System.getProperty("os.name"));
-            }
-        } catch (InterruptedException | IOException e) {
-            throw new RuntimeException(e);
-        } finally {
-            if (process != null) {
-                process.destroy();
-            }
-        }
     }
 
-    public AADB2CSeleniumITHelper(Class<?> appClass, Map<String, String> properties) throws InterruptedException {
+    public AADB2CSeleniumITHelper(Class<?> appClass, Map<String, String> properties) {
         userEmail = AADB2CTestUtils.AAD_B2C_USER_EMAIL;
         userPassword = AADB2CTestUtils.AAD_B2C_USER_PASSWORD;
+        this.appClass = appClass;
+        this.properties = properties;
+    }
+
+    @Override
+    public void appInit(){
         app = new AppRunner(appClass);
         DEFAULT_PROPERTIES.forEach(app::property);
         properties.forEach(app::property);
         this.app.start();
-        setDriver();
-    }
-
-    public void quitDriver() {
-        try {
-            this.driver.quit();
-        } catch (Exception e) {
-            this.driver = null;
-        }
     }
 
-    private void setDriver() {
-        if (driver == null) {
-            ChromeOptions options = new ChromeOptions();
-            options.addArguments("--headless");
-            options.addArguments("--incognito", "--no-sandbox", "--disable-dev-shm-usage");
-            this.driver = new ChromeDriver(options);
-            wait = new WebDriverWait(driver, 5);
-        }
-    }
-
-    public void signIn(String userFlowName) throws InterruptedException {
+    public void signIn() {
         driver.get(app.root());
-        wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[type='submit']")));
-        driver.findElement(By.id("email")).sendKeys(userEmail);
-        driver.findElement(By.id("password")).sendKeys(userPassword);
-        driver.findElement(By.cssSelector("button[type='submit']")).click();
+        wait.until(presenceOfElementLocated(By.id("email"))).sendKeys(userEmail);
```

Comment:
If we already have the clickable submit button, should we use wait.until() here to check the presence of other elements?

Reply:
This is more like a code style discussion.

Reply:
After discussed with @backwind1233, we'll choose wait.until().